### PR TITLE
Merge back 2.2

### PR DESCRIPTION
**Pull request by kprokofi**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by eunwoosh — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

**Review by sovrasov — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

**Comment by kprokofi** [comment]
I will update this PR after RC14 will be finalized

**Review by eunwoosh — DISMISSED** [review]

**Review by sovrasov — DISMISSED** [review]

**Comment by eugene123tw** [comment]
@kprokofi Could you add a line to `src/otx/core/data/transform_libs/torchvision.py`
image
This should fix the index error.

**Review by eugene123tw — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Comment by sovrasov** [comment]
Now the failure is: 
```
setattr(inputs, key, attr[keep])
IndexError: boolean index did not match indexed array along dimension 0; dimension is 3 but corresponding boolean dimension is 2
```
@kprokofi could you disable the failing RTMDet test? We should keep in mind that the model is dysfunctional for now

**Comment by eugene123tw** [comment]
> Now the failure is:
> 
> ```
> setattr(inputs, key, attr[keep])
> IndexError: boolean index did not match indexed array along dimension 0; dimension is 3 but corresponding boolean dimension is 2
> ```
> 
> @kprokofi could you disable the failing RTMDet test? We should keep in mind that the model is dysfunctional for now

@sovrasov @kprokofi yeah, that's disable inst seg recipes for now. I think there's some bug in FilterAnnotations. I will open a PR to fix it.

**Review by eugene123tw — DISMISSED** [review]

**Review by sovrasov — DISMISSED** [review]

File: src/otx/recipe/instance_segmentation/rtmdet_inst_tiny.yaml [hunk 1]
```diff
@@ -87,9 +87,6 @@ overrides:
             max_cached_images: 10
             random_pop: false
             prob: 0.5
```
Comment (eugene123tw) [inline src/otx/recipe/instance_segmentation/rtmdet_inst_tiny.yaml:89]: @kprokofi I think we need to add `is_numpy_to_tvtensor: true` under `CachedMixUp`. Could you check? Thanks!

Comment (kprokofi) [reply src/otx/recipe/instance_segmentation/rtmdet_inst_tiny.yaml:89]: Thanks!

**Review by eugene123tw — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]

**Pull request merged**